src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -271,36 +669,49 @@ private RouteFW resolveTarget(
         private final long routeId;
         private final long initialId;
         private final long replyId;
+        private final long affinity;
+        private final long budgetId;
+
         private final Map<String, MqttSubscribeStream> subscribers;
         private final Map<String, MqttPublishStream> publishers;
         private final Int2ObjectHashMap<Subscription> subscriptionsByPacketId;
-        private long authorization;
 
         private int initialBudget;
         private int initialPadding;
-        private int replyBudget;
         private int replyPadding;
+        private int replyBudget;
+
+        private int decodeSlot = NO_SLOT;
+        private int decodeSlotLimit;
 
-        private long decodeTraceId;
-        private DecoderState decodeState;
-        private int slotIndex = NO_SLOT;
-        private int slotLimit;
-        private long affinity;
+        private int encodeSlot = NO_SLOT;
+        private int encodeSlotOffset;
+        private long encodeSlotTraceId;
+        private int encodeSlotMaxLimit = Integer.MAX_VALUE;
+
+        private MqttServerDecoder decoder;
+
+        private boolean connected;
 
         private MqttServer(
             MessageConsumer network,
             long routeId,
             long initialId,
-            long replyId)
+            long replyId,
+            long affinity,
+            long budgetId)
         {
             this.network = network;
             this.routeId = routeId;
             this.initialId = initialId;
             this.replyId = replyId;
-            this.decodeState = this::decodeConnectPacket;
+            this.affinity = affinity;
+            this.budgetId = budgetId;
+            this.decoder = decodePacketType;
             this.subscribers = new HashMap<>();
             this.publishers = new HashMap<>();
             this.subscriptionsByPacketId = new Int2ObjectHashMap<>();
+            this.connected = false;
```

Comment:
No need for explicit assignment to language defined default - this line should be removed.